src/msw/webview_edge.cpp

```diff
@@ -0,0 +1,777 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        source/msw/webview_edge.cpp
+// Purpose:     wxMSW Edge Chromium wxWebView backend implementation
+// Author:      Markus Pingel
+// Created:     2019-12-15
+// Copyright:   (c) 2019 wxWidgets development team
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#if defined(__BORLANDC__)
+#pragma hdrstop
+#endif
+
+#include "wx/msw/webview_edge.h"
+
+#if wxUSE_WEBVIEW && wxUSE_WEBVIEW_EDGE
+
+#include "wx/filename.h"
+#include "wx/module.h"
+#include "wx/stdpaths.h"
+#include "wx/thread.h"
+#include "wx/private/jsscriptwrapper.h"
+
+#include <wrl/event.h>
+
+using namespace Microsoft::WRL;
+
+wxIMPLEMENT_DYNAMIC_CLASS(wxWebViewEdge, wxWebView);
+
+#define WX_ERROR2_CASE(error, wxerror) \
+        case error: \
+            event.SetString(#error); \
+            event.SetInt(wxerror); \
+            break;
+
+int wxWebViewEdge::ms_isAvailable = -1;
+// WebView2Loader typedefs
+typedef HRESULT (__stdcall *PFNWXCreateWebView2EnvironmentWithDetails)(
+    PCWSTR browserExecutableFolder,
+    PCWSTR userDataFolder,
+    PCWSTR additionalBrowserArguments,
+    IWebView2CreateWebView2EnvironmentCompletedHandler* environment_created_handler);
+typedef HRESULT(__stdcall *PFNWXGetWebView2BrowserVersionInfo)(
+    PCWSTR browserExecutableFolder,
+    LPWSTR* versionInfo);
+
+PFNWXCreateWebView2EnvironmentWithDetails wxCreateWebView2EnvironmentWithDetails = NULL;
+PFNWXGetWebView2BrowserVersionInfo wxGetWebView2BrowserVersionInfo = NULL;
+
+wxDynamicLibrary wxWebViewEdge::ms_loaderDll;
+
+bool wxWebViewEdge::IsAvailable()
+{
+    if (ms_isAvailable == -1)
+    {
+        if (!Initialize())
+            ms_isAvailable = 0;
+        else
+            ms_isAvailable = 1;
+    }
+
+    return (ms_isAvailable == 1);
+}
+
+bool wxWebViewEdge::Initialize()
+{
+#define RESOLVE_LOADER_FUNCTION(type, funcname)                        \
+    wx##funcname = (type)ms_loaderDll.GetSymbol(wxT(#funcname));       \
+    if ( !wx##funcname )                                               \
+        return false
+
+    // WebView2 is only available for Windows 7 or newer
+    if (!wxCheckOsVersion(6, 1))
+        return false;
```

Comment:
My idea was blocking XP users, but sure if the WebView2Loader.dll was marked as Win7+ ist would fail loading and otherwise the `GetVersionInfo` call will fail. Will remove.